docs/shared/concepts/typescript-project-linking.md

```diff
@@ -0,0 +1,277 @@
+
+{% github-repository title="TypeScript Project References Benchmark" url="https://github.com/jaysoo/typecheck-timings" /%}
+
+Here are the baseline typecheck task performance results.
+
+```text
+Typecheck without using project references: 186 seconds, max memory 6.14 GB
+```
+
+Using project references allows the TypeScript compiler to individually check the types for each project and store the results of that calculation in a `.tsbuildinfo` file for later use. Because of this, the TypeScript compiler does not need to load the entire codebase into memory at the same time, which you can see from the decreased memory usage on the first run with project references enabled.
+
+```text
+Typecheck with project references first run: 175 seconds, max memory 945 MB
+```
+
+Once the `.tsbuildinfo` files have been created, subsequent runs will be much faster.
+
+```text
+Typecheck with all `.tsbuildinfo` files created: 25 seconds, max memory 429 MB
+```
+
+Even if some projects have been updated and individual projects need to be type checked again, the TypeScript compiler can still use the cached `.tsbuildinfo` files for any projects that were not affected. This is very similar to the way Nx's caching and affected features work.
+
+```text
+Typecheck (1 pkg updated): 36.33 seconds, max memory 655.14 MB
+Typecheck (5 pkg updated): 48.21 seconds, max memory 702.96 MB
+Typecheck (25 pkg updated): 65.25 seconds, max memory 666.78 MB
+Typecheck (100 pkg updated): 80.69 seconds, max memory 664.58 MB
+Typecheck (1 nested leaf pkg updated): 26.66 seconds, max memory 407.54 MB
+Typecheck (2 nested leaf pkg updated): 31.17 seconds, max memory 889.86 MB
+Typecheck (1 nested root pkg updated): 26.67 seconds, max memory 393.78 MB
+```
+
+These performance benefits will be more noticeable for larger repositories, but even small code bases will see some benefits.
+
+### Local TypeScript Path Aliases
+
+When using project references, you can not define path aliases in the root `tsconfig.base.json` file because TypeScript does not merge the path aliases when doing the typecheck calculation, but you can set path aliases in an individual project's `tsconfig.app.json` file. For instance, you could define paths like this in an application's tsconfig file.
+
+```jsonc {% fileName="/apps/my-remix-app/tsconfig.app.json" %}
+{
+  "compilerOptions": {
+    "paths": {
+      "#app/*": ["./app/*"],
+      "#tests/*": ["./tests/*"],
+      "@/icon-name": [
+        "./app/components/ui/icons/name.d.ts",
+        "./types/icon-name.d.ts"
+      ]
+    }
+  }
+}
+```
+
+## Project Linking with TypeScript Path Aliases
+
+{% callout type="warning" title="Incompatible with TypeScript Project References" %}
+You can not use TypeScript project references with this style of project linking.
```

Comment:
this is technically not correct.

You can use TS path aliases with project references. The former is for the name resolution while the latter defines how TS creates "programs" for type checking/compilation.

Now, while possible, it's not what we recommend for sure. We should somewhere mention though that the recommendation goes:

- Workspaces + proj references (preferred for new workspaces)
- TS path aliases (how Nx does it now and will still be valid for a long time)